Frameworks/UIKit/UILabel.mm

```diff
@@ -44,61 +44,70 @@ @implementation UILabel {
     UILineBreakMode _lineBreakMode;
     BOOL _adjustFontSize;
     float _minimumFontSize;
-    float _originalFontSize;
     int _numberOfLines;
     BOOL _isDisabled;
     BOOL _isHighlighted;
     UIBaselineAdjustment _baselineAdjustment;
     StrongId<WXCTextBlock> _textBlock;
 }
 
-- (void)adjustFontSizeToFit {
+- (void)_adjustFontSizeToFit {
     if (_numberOfLines == 0) {
-        [self adjustTextLayerSize];
+        [self _adjustTextLayerSize];
         return;
     }
 
-    float curFontSize = _originalFontSize;
-    _font = [_font fontWithSize:curFontSize];
-
-    CGRect rect;
-    rect = [self bounds];
-    if (rect.size.width == 0.0f || rect.size.height == 0.0f) {
+    CGRect rect = [self bounds];
+    if (rect.size.width == 0.0f || rect.size.height == 0.0f || _text == nil) {
         return;
     }
 
-    while (curFontSize > _minimumFontSize && curFontSize > 0.0f) {
-        CGSize size = CGSizeZero;
+    // MaxFitFontSize records currently found maximum font size that fits the bounds,
+    // It starts with _minimumFontSize because it is smallest font size allowed by the UILabel
+    float maxFitFontSize = _minimumFontSize;
+
+    float curFontSize = [_font pointSize];
+    // Start our trial with fonts bigger than _minimumFontSize
+    if (curFontSize < maxFitFontSize) {
+        curFontSize = 2.0 * maxFitFontSize;
```

Comment:
> curFontSize = 2.0 * maxFitFontSize; [](start = 7, length = 36)

Curious why not start at std::max<float>(curFontSize, maxFitFontSize), vs. starting with 2x maxFitFontSize?